api/data/NMBS/vehicleinformation.php

```diff
@@ -61,7 +68,7 @@ private static function getServerData($id, $lang)
         $scrapeURL = 'http://www.belgianrail.be/jp/sncb-nmbs-routeplanner/trainsearch.exe/'.$lang.'ld=std&seqnr=1&ident=at.02043113.1429435556&';
         $id = preg_replace("/[a-z]+\.[a-z]+\.([a-zA-Z0-9]+)/smi", '\\1', $id);
 
-        $post_data = 'trainname='.$id.'&start=Zoeken&selectDate=oneday&date='.date('d%2fm%2fY').'&realtimeMode=Show';
+        $post_data = 'trainname='.$id.'&start=Zoeken&selectDate=oneday&date='.DateTime::createFromFormat('dmy', $date)->format('d%2fm%2fY').'&realtimeMode=Show';
```

Comment:
I think you need to add a `%2f`between `d` and `m` and between `m` and `Y` (mind the capital Y). This would be correct: `DateTime::createFromFormat('d%2fm%2fY', $date)`.

Did you test your solution whether the data is really correct? Because to me it seems like the URL you created is just going to return today's train schedule as it won't be able to parse your date?